### ISSUE TITLE
Vocab labels empty

### ISSUE THREAD

**Issue by nihiluis** [allenai/allennlp#2038]
**Describe the bug**
I use this reader
```python
class MRDatasetReader(DatasetReader):
    def __init__(self,
             tokenizer: Tokenizer = None,
             token_indexers: Dict[str, TokenIndexer] = None,
             lazy: bool = False) -> None:
        super().__init__(lazy)

        self._tokenizer = tokenizer or WordTokenizer()
        self._token_indexers = token_indexers or {'tokens': SingleIdTokenIndexer()}
    
    def _read(self, file_path):
        with open(cached_path(file_path), "r") as data_file:
            df = pd.read_csv(file_path, dtype={"PhraseId": np.float, "Sentiment": np.float, "SentenceId": np.float})
            #print("count = ", len(train.where(train['Sentiment'] == 0).dropna()))
            for i, item in df.iterrows():
                phrase_id = item["PhraseId"]
                sentence_id = item["SentenceId"]
                phrase = item["Phrase"]
                sentiment = item["Sentiment"]
                yield self.text_to_instance(phrase_id, sentence_id, phrase, sentiment)
            
    def text_to_instance(self, phrase_id, sentence_id, phrase, sentiment) -> Instance:
        tokenized_phrase = self._tokenizer.tokenize(phrase)
        
        phrase_field = TextField(tokenized_phrase, self._token_indexers)
        #phrase_id_field = MetadataField(phrase_id)
        #sentence_id_field = MetadataField(sentence_id)
        fields = {
            "phrase": phrase_field
        }
        
        #print(f"sentiment = {sentiment} | sentiment-1 = {sentiment-1}")
        fields["labels"] = LabelField(sentiment, skip_indexing=True)
            
        return Instance(fields)
```
sentiment is an int.

now when I load my vocabulary `vocab = Vocabulary.from_instances(val_dataset)`

and print this: `vocab._token_to_index["labels"]`

it returns {}

this is a problem ...

now if i replace the LabelField like this: `fields["labels"] = LabelField(str(int(sentiment)))`

it returns then this: `{'2': 0, '3': 1, '1': 2, '4': 3, '0': 4}` (for this print: `vocab._token_to_index["labels"]`)

**To Reproduce**
Steps to reproduce the behavior

**Expected behavior**
A clear and concise description of what you expected to happen. Ok

skip_indexing work fine .. i need to get something like {0:0, 1:1, 2:2, etc..}. i dont know but definitely not empty like {}. i need labels to pass to forward with full size, not just 0 when I call `print("vocab size output =", vocab.get_vocab_size("labels"))`. when I do it without skip_indexing

**System (please complete the following information):**
 - OS: [e.g. OSX, Linux] kaggle
 - Python version: [if it's not 3.6.1 or later, that's probably your problem] its the kaggle version
 - AllenNLP version: [e.g. v0.7.0, or "I installed from master"] 0.7.1
 - PyTorch version: (if you installed it yourself) 0.4.1

**Additional context**
Yes

i load vocab like this: `vocab = Vocabulary.from_instances(train_dataset + val_dataset)`

before i add token_indexers to datasetreader

```
token_indexers = { 
    "tokens": SingleIdTokenIndexer(lowercase_tokens=True),
    "elmo": ELMoTokenCharactersIndexer(namespace="elmo"),
    "token_characters": TokenCharactersIndexer(character_tokenizer=CharacterTokenizer(byte_encoding="utf-8", start_tokens=[259], end_tokens=[260]))
}
```

**Comment by matt-gardner** [comment]
This looks to me like it's working as intended.  I'm not sure what the problem is.  If you tell the `LabelField` to skip indexing, we won't create a vocabulary (this is intentional), and if you _do_ tell it to index, you'll get a mostly-arbitrary mapping from your labels to integers, which you can reverse to get back your original labels (this is also intentional).  What's the issue?

**Comment by nihiluis** [comment]
Ah, sorry. I thought this was necessary since one example required the labels to have vocab (to determine the output features of the final linear layer), newbie mistake.
